src/renderer.ts

```diff
@@ -272,8 +272,14 @@ export function getAllDependencies (ids: Set<Identifier>, rendererContext: Rende
 
   for (const id of ids) {
     const deps = getModuleDependencies(id, rendererContext)
-    Object.assign(allDeps.scripts, deps.scripts)
-    Object.assign(allDeps.styles, deps.styles)
+    const isDynamic = rendererContext.clientManifest[id]?.isDynamicEntry
+    if (isDynamic) {
+      Object.assign(allDeps.prefetch, deps.styles)
+      Object.assign(allDeps.prefetch, deps.scripts)
```

Comment:
Note for later: We need a feature flag to control `shouldPrefetch` or globally disable prefetch for dynamic chunks. On bigger projects overprefetch has negative perf

Reply:
Made the tracker #20